### PR TITLE
add guard against string value being treated as array

### PR DESCRIPTION
**Pull request by serkan-paystack**
This PR addresses https://github.com/Valve/fingerprintjs2/issues/432 where canvas might possibly be throwing and returning an error string instead of an expected error.

The hotfix simply guards against the value not being an array and instead continues with the program flow.

This does not necessarily fix the underlying root cause, merely ensures it does not break full execution.

### CODE REVIEW

**Comment by serkanatpaystack** [comment]
Hey @jonashaag pinging you here as you're already familiar with this use case.

**Comment by nebi-frame** [comment]
why are we not merging this ? PR seems promising and seems passed all the checks?

**Comment by serkanatpaystack** [comment]
@nebi-frame fwiw we migrated away from the 1.8 API and embraced the new api.

**Comment by Valve** [comment]
@jonashaag please review if you have time, alternatively I'll review later this week

File: fingerprint2.js [hunk 1]
```diff
@@ -1388,14 +1388,9 @@
...
-          newComponents.push({key: component.key, value: component.value.join('~')})
+          if (component.value && typeof component.value.join === 'function') newComponents.push({key: component.key, value: component.value.join('~')})
         } else if (['sessionStorage', 'localStorage', 'indexedDb', 'addBehavior', 'openDatabase'].indexOf(component.key) !== -1) {
-          if (component.value) {
-            newComponents.push({key: component.key, value: 1})
-          } else {
-            // skip
-            continue
-          }
+          if (component.value) newComponents.push({key: component.key, value: 1})
```
Comment (jonashaag) [inline fingerprint2.js]: @Valve I'd undo this change.

File: fingerprint2.js [hunk 2]
```diff
@@ -1388,14 +1388,9 @@
               return [p[0], p[1], mimeTypes].join('::')
             })})
         } else if (['canvas', 'webgl'].indexOf(component.key) !== -1) {
-          newComponents.push({key: component.key, value: component.value.join('~')})
+          if (component.value && typeof component.value.join === 'function') newComponents.push({key: component.key, value: component.value.join('~')})
```
Comment (jonashaag) [inline fingerprint2.js]: @Valve I think this should have an else case (just add `value: component.value`)

Comment (Valve) [reply fingerprint2.js]: sounds good, thanks

**Comment by Valve** [comment]
this has been implemented in https://github.com/Valve/fingerprintjs2/commit/d9d4b0dd266d57d53a06e05aa6356a1513fcf7da
closing.
